### ISSUE TITLE
COULD NOT FIND session.key - Timed Out.

### ISSUE THREAD

**Issue by green-butterfly** [zerosum0x0/koadic#146]
Hey All.

**Attacker**
OS = Kali Linux 2021.1
Kernel = 5.10.0-kali4-amd64
Environment = Laptop (main os)

**Target**
OS = Windows 10 pro
build = 19041.804
Antivirus = off
Firewall = off
Environment = virtualbox 6.1.18
Host = Ubuntu 20.04.2 LTS

**Issue**
Could not find session key.

```
(koadic: sta/js/mshta)# verbose 1
[*] Verbose mode: On
[v] handler::handle() - Incoming HTTP from ('192.168.1.204', 49705)
[v] handler::parse_params() - Host header present: 192.168.1.100:9999
[v] handler::handle_new_session()
[+] Zombie 1: Staging new connection (192.168.1.204) on Stager 0
[v] handler::reply() - sending status 200 with 27031 bytes to ('192.168.1.204', 49705)
[v] handler::handle() - Incoming HTTP from ('192.168.1.204', 49706)
[v] handler::parse_params() - COULD NOT FIND session.key = 2aefff9f55a745e0ab9decf81f354c8a;DZGSZWXOEL=;
[v] handler::reply() - sending status 404 with 0 bytes to ('192.168.1.204', 49706)
[v] handler::handle() - Incoming HTTP from ('192.168.1.204', 49707)
[v] handler::parse_params() - COULD NOT FIND session.key = 2aefff9f55a745e0ab9decf81f354c8a;DZGSZWXOEL=;\..\..\..\./mshtml,RunHTMLApplication
[v] handler::reply() - sending status 404 with 0 bytes to ('192.168.1.204', 49707)
[!] Zombie 1: Timed out.
(koadic: sta/js/mshta)# 
```

Please advise.
Thanks.

**Comment by TheNaterz** [comment]
Looks like the issue is in urllib.parse.parse_qs for later versions of Python 3. https://docs.python.org/3/library/urllib.parse.html

We take advantage of the fact that urllib.parse_qs, for versions of Python < 3.9.2, allows for both ';' and '&' characters as separators. Our payloads use ';' as a separator. This behavior has apparently changed in later versions of Python 3 to add a separator parameter and defaulting to '&'. I'm not sure how the payloads would be affected if I changed the separator, and it's frustrating that Python has decided to break this functionality.

I'm gonna remove the dependency on urllib here and write a simple parser for the query string.

**Comment by TheNaterz** [comment]
Pushed a fix bdce981e2759e04ebf98f5608a04365a3d818e59. Try pulling and running again.
